src/scripts/index.js

```diff
@@ -0,0 +1,302 @@
+  cardLikeBtn.addEventListener("click", (evt) => {
+    api
+      .changeLikeStatus(data._id, data.isLiked)
+      .then((updatedCard) => {
+        console.log(updatedCard);
+        evt.target.classList.toggle("card__like-btn_liked");
+      })
+      .catch((err) => console.error(`Error updating like status: ${err}`));
+  });
+
+  cardDeleteBtn.addEventListener("click", () => {
+    selectedCard = cardElement;
+    selectedCardID = data._id;
+    openModal(deleteModal);
+  });
+
+  return cardElement;
+}
+
+function handleEscClose(evt) {
+  if (evt.key === "Escape") {
+    const openModal = document.querySelector(".modal_opened");
+    if (openModal) {
+      closeModal(openModal);
+    }
+  }
+}
+
+function openModal(modal) {
+  if (!modal.classList.contains("modal_opened")) {
+    modal.classList.add("modal_opened");
+    document.addEventListener("keydown", handleEscClose);
+  }
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscClose);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtontext(submitBtn, true, "Save", "Saving...");
+
+  // Call API to update user info
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch((error) => {
+      console.error("Error updating profile:", error);
+    })
+    .finally(() => {
+      setButtontext(submitBtn, false, "Save", "Saving...");
+    });
+}
+
+//TODO - implement loading text for all other form submissions
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+
+  setButtontext(submitBtn, true, "Save", "Saving...");
+
+  const cardData = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  };
+
+  api
+    .addCard(cardData)
+    .then((newCard) => {
+      cardsList.prepend(getCardElement(newCard)); // Prepend new card
+      cardForm.reset(); // Reset form
+      resetValidation(cardForm, settings); // Reset validation
+      closeModal(cardModal); // Close modal
+    })
+    .catch((err) => console.error(`Error adding card: ${err}`))
+    .finally(() => setButtontext(submitBtn, false, "Save", "Saving...")); // Reset button text
+}
+
+function handleDeleteCard(cardElement, cardID) {
+  selectedCard = cardElement;
+  selectedCardID = cardID;
+  openModal(deleteModal);
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  resetValidation(editFormElement, settings);
+  openModal(editModal);
+});
+
+cardModalBtn.addEventListener("click", () => {
+  openModal(cardModal);
+});
+
+editModalClosedBtn.addEventListener("click", () => {
```

Comment:
**Could be improved**

You can make a universal handler for any close buttons.

It will look something like this:

```js
// Find all close buttons
const closeButtons = document.querySelectorAll('.modal__close');

closeButtons.forEach((button) => {
  // Find the closest popup only once
  const popup = button.closest('.modal');
  // Set the listener
  button.addEventListener('click', () => closePopup(popup));
});



```

That’s why we make universal css classes for similar elements: here it’s `modal__close` for any close button.

So now you can add 100 modals: their close buttons will be handled automatically.